src/ffi/socket.rs

```diff
@@ -305,6 +309,22 @@ impl Socket {
             join_handle: AtomicTake::new(join_handle),
         }))
     }
+    #[cfg(not(feature = "uniffi"))]
+    pub fn spawn(mut url: Url) -> Result<Arc<Self>, SpawnError> {
+        Self::spawn_actual(url)
+    }
+}
+#[cfg_attr(
+    feature = "uniffi",
+    uniffi::export
+)]
+impl Socket {
+    /// Spawns a new [Socket] that must be [Socket::connect]ed.
+    #[cfg(feature = "uniffi")]
+    #[uniffi::constructor]
```

Comment:
Shouldn't this be `cfg_attr` instead of `cfg`?  `spawn` needs to still exist even with `uniffi` turned off.

Reply:
I spent a lot of time trying to do this with one `cfg_attr` but it didn't work. I ended up looking at the `cargo-expanded` out and I think the order of macro executation isn't in our favor. I had to do this here and with `Topic::from_string`.

Reply:
> spawn needs to still exist even with uniffi turned off.

I verified (and have since added it to CI) that it builds the tests with uniffi turned off.